activerecord/lib/active_record/persistence.rb

```diff
@@ -307,18 +307,19 @@ def instantiate(attributes, column_types = {}, &block)
       # it is preferred to use {update_all}[rdoc-ref:Relation#update_all]
       # for updating all records in a single query.
       def update(id = :all, attributes)
+        if ActiveRecord::Base === id || id.is_a?(Array) && id.any? { |one_id| ActiveRecord::Base === one_id }
```

Comment:
How do you feel about using `.is_a?` for all 3 of these instead of using `.is_a?` and `===`?

```suggestion
        if id.is_a?(ActiveRecord::Base) || id.is_a?(Array) && id.any? { |one_id| one_id.is_a?(ActiveRecord::Base) }
```